src/App.js

```diff
@@ -1,85 +1,208 @@
-import React from 'react';
+import React, { PureComponent } from 'react';
+import { v4 } from 'uuid';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
 
-function App() {
-  return (
-    <section className="todoapp">
-      <header className="header">
-        <h1>todos</h1>
+const todos = [
+  {
+    id: v4().substr(0, 4),
+    label: 'drink a coffee',
+    completed: false,
+  },
+  {
+    id: v4().substr(0, 4),
+    label: 'get the world masters',
+    completed: false,
+  },
+  {
+    id: v4().substr(0, 4),
+    label: 'find myself',
+    completed: false,
+  },
+  {
+    id: v4().substr(0, 4),
+    label: 'make somebody happy',
+    completed: false,
+  },
+];
 
-        <input
-          className="new-todo"
-          placeholder="What needs to be done?"
+export class App extends PureComponent {
+  state = {
+    todoList: todos,
+    isSorted: false,
+    sortedTodos: [],
+  };
+
+  componentDidMount() {
+    if (localStorage.getItem('todos')) {
+      this.setState(prevState => ({
+        todoList: [
+          ...JSON.parse(localStorage.getItem('todos')),
+        ],
+      }));
+    }
+  }
+
+  componentDidUpdate() {
+    localStorage.setItem('todos', JSON.stringify(this.state.todoList));
+  }
+
+  createTodo = todo => ({
```

Comment:
this is not the place where you should do this, a new todo should be created right there where you call the corresponding method